snapshot-labs/snapshot-sidekick
# fix: prevent double minting

## 🧿 Current issues / What's wrong ?

Second part of the following pitch: https://www.notion.so/snapshotlabs/NFT-claimer-minting-constrains-de9705b71aca4b1aa57ff62b47d1824a

Based on  https://github.com/snapshot-labs/snapshot-sidekick/pull/184

## 🚧 Changes

- Prevent minting when the address has already minted from the same proposal

## 🛠️ Tests

- Run a local instance of this branch (Copy and set the missing env var from /test/.env.test)
- Run a local instance of snapshot https://github.com/snapshot-labs/snapshot/pull/3885 (set the `VITE_SIDEKICK_URL` in .env to your local sidekick instance)
- On a proposal you have voted and minted, if you try to mint, it should return an error
- On a proposal you have voted but not minted, if you try to mint, it should mint

As of now, the UI only return "Invalid data submitted" message on error. UI needs some refactoring to check if user already voted, and show am error another message, instead of starting the mint and failing from sidekick validation.
